File: sdk/src/main/scala/com/horizen/vrf/VRFKeyGenerator.scala
package com.horizen.vrf

// See https://tools.ietf.org/id/draft-goldbe-vrf-01.html#rfc.section.2 as functions description
object VRFKeyGenerator {
  def generate(seed: Array[Byte]): (VRFSecretKey, VRFPublicKey) = {
    val secret = new VRFSecretKey(Array.fill(VRFSecretKey.length)(seed.deep.hashCode().toByte))
    val public = new VRFPublicKey(Array.fill(VRFPublicKey.length)(seed.deep.hashCode().toByte))

    (secret, public)
  } // jni call to Rust impl
}

File: sdk/src/main/java/com/horizen/vrf/VRFKeyGenerator.java
package com.horizen.vrf;

import com.horizen.utils.Pair;

public class VRFKeyGenerator {

  public static Pair<VRFSecretKey, VRFPublicKey> generate(byte[] seed) {
    return new Pair(new VRFSecretKey(new byte[32]) , new VRFPublicKey(new byte[32]));
  } // jni call to Rust impl
}
